packages/ui-components/src/components/text-field/text-field.tsx

```diff
@@ -0,0 +1,65 @@
+import { Component, Event, EventEmitter, h, Host, Prop } from '@stencil/core';
+import { EInputFieldType, IErrorMsgs } from './text-field.types';
+
+@Component({
+	tag: 'kv-text-field',
+	styleUrl: 'text-field.scss',
+	shadow: true,
+})
+export class KvTextField {
+	/** (optional) Text field tyoe */
+	@Prop({ reflect: true }) type!: EInputFieldType;
+	/** (optional) Text field label */
+	@Prop({ reflect: true }) label: string;
+	/** (optional) Text field input name */
+	@Prop({ reflect: true }) inputName: string;
+	/** (optional) Text field place holder */
+	@Prop({ reflect: true }) placeholder: string;
+	/** (optional) Text field disabled */
+	@Prop({ reflect: true }) disabled = false;
+	/** (optional) Text field required */
+	@Prop({ reflect: true }) required = false;
+	/** (optional) Text field loading state */
+	@Prop({ reflect: true }) loading = false;
+	/** (optional) Text field help text */
+	@Prop({ reflect: true }) helpText: string;
+	/** (optional) Text field error messages */
+	@Prop({ reflect: true }) errorMessages: IErrorMsgs;
+	/** (optional) Text field value */
+	@Prop({ reflect: true }) value: string;
+
+	/** Emitted when text field's value changes */
+	@Event() textChange: EventEmitter<InputEvent>;
+
+	private onTextChange = (event: InputEvent) => {
```

Comment:
In our previous library we were using the blur event to dispatch a change on the text input, I don't know if this behavior is to be kept or if your implementation we stick with. @miguelpinto98 @djcouto @ricardogoncalves89 @Insatisfeito